ghost/core/test/e2e-browser/utils/e2e-browser-utils.js

```diff
@@ -250,64 +250,68 @@ const createTier = async (page, {name, monthlyPrice, yearlyPrice, trialDays}, en
  */
 
 const createOffer = async (page, {name, tierName, offerType, amount, discountType = null, discountDuration = 3}) => {
-    await page.goto('/ghost');
-    await page.locator('[data-test-nav="settings"]').click();
-
-    // Keep offer names unique & <= 40 characters
-    let offerName = `${name} (${new ObjectID().toHexString().slice(0, 40 - name.length - 3)})`;
-    // Tiers request can take time, so waiting until there is no connections before interacting with them
-    await page.waitForLoadState('networkidle');
+    let offerName;
+    let offerLink;
+    await test.step('Create an offer', async () => {
+        await page.goto('/ghost');
+        await page.locator('[data-test-nav="settings"]').click();
 
-    const hasExistingOffers = await page.getByTestId('offers').getByRole('button', {name: 'Manage offers'}).isVisible();
-    const isCTA = await page.getByTestId('offers').getByRole('button', {name: 'Add offer'}).isVisible();
-    // Archive other offers to keep the list tidy
-    // We only need 1 offer to be active at a time
-    // Either the list of active offers loads, or the CTA when no offers exist
-    if (hasExistingOffers && !isCTA) {
-        await page.getByTestId('offers').getByRole('button', {name: 'Manage offers'}).click();
-
-        // Selector for the elements with data-testid 'offer-item'
-        // const offerItemsSelector = '[data-testid="offer-item"]';
-        await page.getByTestId('offer-item').nth(0).click();
-        await page.getByRole('button', {name: 'Archive offer'}).click();
-
-        const confirmModal = await page.getByTestId('confirmation-modal');
-        await confirmModal.getByRole('button', {name: 'Archive'}).click();
-    }
+        // Keep offer names unique & <= 40 characters
+        offerName = `${name} (${new ObjectID().toHexString().slice(0, 40 - name.length - 3)})`;
```

Comment:
there is a potential issue with the code: it assumes that name.length is less than or equal to 37. If the length of name exceeds this limit, the generated string might be longer than 40 characters.

To make the code more robust, you can add a check to ensure that the resulting string does not exceed the desired length.

`const maxOfferNameLength = 40;

let offerName = `${name} (${new ObjectID().toHexString().slice(0, maxOfferNameLength - name.length - 3)})`;

// Ensure the offerName does not exceed the maximum length
if (offerName.length > maxOfferNameLength) {
    offerName = offerName.slice(0, maxOfferNameLength);
}`